src/System.Net.HttpListener/src/System/Net/HttpListenerRequest.Unix.cs

```diff
@@ -1,268 +1,667 @@
+                     c != ':' && c != '\\' && c != '"' && c != '/' && c != '[' &&
+                     c != ']' && c != '?' && c != '=' && c != '{' && c != '}'))
+                    continue;
+
+                _context.ErrorMessage = "(Invalid verb)";
+                return;
             }
-        }
 
-        public string ContentType
-        {
-            get
+            _raw_url = parts[1];
+            if (parts[2].Length != 8 || !parts[2].StartsWith("HTTP/"))
             {
-                throw new PlatformNotSupportedException();
+                _context.ErrorMessage = "Invalid request line (version).";
+                return;
             }
-        }
 
-        public NameValueCollection Headers
-        {
-            get
+            try
             {
-                throw new PlatformNotSupportedException();
+                _version = new Version(parts[2].Substring(5));
+                if (_version.Major < 1)
+                    throw new Exception();
+            }
+            catch
+            {
+                _context.ErrorMessage = "Invalid request line (version).";
+                return;
             }
         }
 
-        public string HttpMethod
+        private void CreateQueryString(string query)
         {
-            get
+            if (query == null || query.Length == 0)
             {
-                throw new PlatformNotSupportedException();
+                _query_string = new NameValueCollection(1);
+                return;
             }
-        }
 
-        public Stream InputStream
-        {
-            get
+            _query_string = new NameValueCollection();
+            if (query[0] == '?')
+                query = query.Substring(1);
+            string[] components = query.Split('&');
+            foreach (string kv in components)
             {
-                throw new PlatformNotSupportedException();
+                int pos = kv.IndexOf('=');
+                if (pos == -1)
+                {
+                    _query_string.Add(null, WebUtility.UrlDecode(kv));
+                }
+                else
+                {
+                    string key = WebUtility.UrlDecode(kv.Substring(0, pos));
+                    string val = WebUtility.UrlDecode(kv.Substring(pos + 1));
+
+                    _query_string.Add(key, val);
+                }
             }
         }
 
-        public bool IsAuthenticated
+        private static bool MaybeUri(string s)
         {
-            get
-            {
-                throw new PlatformNotSupportedException();
-            }
+            int p = s.IndexOf(':');
+            if (p == -1)
+                return false;
+
+            if (p >= 10)
+                return false;
+
+            return IsPredefinedScheme(s.Substring(0, p));
         }
 
-        public bool IsLocal
+        //
+        // Using a simple block of if's is twice as slow as the compiler generated
```

Comment:
Instead of this comment that could be out of sync (given the change in compiler, native compilation, etc), use a UnitTest/PerfTest.